### PR TITLE
Create CIS check for contained database to have auto closed set to true

### PR DESCRIPTION
**Pull request by tboggiano**
# A New PR

THANK YOU - We love to get PR's and really appreciate your time and help to improve this module

## Accepting a PR

Before we accept the PR - please confirm that you have run the tests locally to avoid our automated build and release process failing. You can see how to do that in our wiki

https://github.com/sqlcollaborative/dbachecks/wiki 

## Please confirm you have 0 failing Pester Tests

[X] There are 0 failing Pester tests

## Changes this PR brings

- CIS check for if contained database property for auto close is set to true

Please add below the changes that this PR covers. It doesnt need an essay just the bullet points :-)

### CODE REVIEW

File: internal/configurations/DbcCheckDescriptions.json [hunk 1]
```diff
@@ -438,5 +438,13 @@
     {
         "UniqueTag": "LatestBuild",
         "Description": "Tests that the sees is SQL Server has the newest Service Packs and CUs installed."
+    },
+    {
+        "UniqueTag": "ContainedDBAutoClose",
+        "Description": "Tests that the sees is contained databases has AUTO CLOSE disabled."
```
Comment (SQLDBAWithABeard) [inline internal/configurations/DbcCheckDescriptions.json]: Tests that each contained database ........

File: internal/configurations/DbcCheckDescriptions.json [hunk 2]
```diff
@@ -438,5 +438,13 @@
...
+    },
+    {
+        "UniqueTag": "ContainedDBSQLAuth",
+        "Description": "Tests that the sees is contained databases has does not use SQL authentication."
```
Comment (SQLDBAWithABeard) [inline internal/configurations/DbcCheckDescriptions.json]: Tests that each contained database ........

File: internal/configurations/configuration.ps1 [hunk 3]
```diff
@@ -58,6 +58,8 @@ Set-PSFConfig -Module dbachecks -Name policy.security.oleautomationproceduresdis
 Set-PSFConfig -Module dbachecks -Name policy.security.remoteaccessdisabled -Value 0 -Initialize -Description "Remote Access should be disabled 0"
 Set-PSFConfig -Module dbachecks -Name policy.security.scanforstartupproceduresdisabled -Validation bool -Value $true -Initialize -Description "Scan For Startup Procedures disabled `$true or enabled `$false"
 Set-PSFConfig -Module dbachecks -Name policy.security.latestbuild -Validation bool -Value $true -Initialize -Description "SQL Server should have the latest SQL build (service packs/CUs) installed"
+Set-PSFConfig -Module dbachecks -Name policy.security.containedbautoclose -Validation bool -Value $true -Initialize -Description "Contained databases should have Auto Close enabled"
+Set-PSFConfig -Module dbachecks -Name policy.security.containeddbsqlauth -Validation bool -Value $false -Initialize -Description "Contained databases should not use SQL authenication"
```
Comment (SQLDBAWithABeard) [inline internal/configurations/configuration.ps1]: Can these configs be altered to be skips for the tests and set to true - we will then write the Invoke-DbcCisCheck which will set all required configs and run the checks

File: internal/configurations/configuration.ps1 [hunk 4]
```diff
@@ -237,6 +237,7 @@ Set-PSFConfig -Module dbachecks -Name skip.instance.oleautomationproceduresdisab
 Set-PSFConfig -Module dbachecks -Name skip.instance.remoteaccessdisabled -Validation bool -Value $false -Initialize -Description "Skip the remote access check"
 Set-PSFConfig -Module dbachecks -Name skip.instance.scanforstartupproceduresdisabled -Validation bool -Value $false -Initialize -Description "Skip the scan for startup procedures disabled check"
 Set-PSFConfig -Module dbachecks -Name skip.instance.latestbuild -Validation bool -Value $false -Initialize -Description "Skip the scan the latest build of SQL Server check"
+Set-PSFConfig -Module dbachecks -Name skip.security.containedbautoclose -Validation bool -Value $false -Initialize -Description "Contained databases should have Auto Close enabled"
```
Comment (SQLDBAWithABeard) [inline internal/configurations/configuration.ps1]: This description needs to explain what the config does ;-)

File: checks/Database.Tests.ps1 [hunk 5]
```diff
@@ -938,6 +938,26 @@ $ExcludedDatabases += $ExcludeDatabase
             }
         }
     }
+    Describe "Contained Database Auto Close" -Tags ContainedDBAutoClose, CIS, $filename {
+        if (Get-DbcConfigValue skip.security.containedbautoclose) {
+            if ($NotContactable -contains $psitem) {
```
Comment (SQLDBAWithABeard) [inline checks/Database.Tests.ps1]: Instead of doing this please can you set $skip to the Get-DbcConfigValue and then -Skip:$Skip to the "It" :-)

Comment (tboggiano) [reply checks/Database.Tests.ps1]: So question, by default it would skip this check right?

Comment (SQLDBAWithABeard) [reply checks/Database.Tests.ps1]: Yes I think so

Comment (tboggiano) [reply checks/Database.Tests.ps1]: I figured out how to check, it does skip it by default so this PR should be good to go.

**Pull request merged**